Jest Testing and Basic Database Communications

VejayPersaud opened the pull request:
Re-Organizes the project to include jest testing and basic database communication and operations

VejayPersaud committed:
Testing database functionality

Added new files and structure

VejayPersaud committed:
Database Processing Initial Setup

 Adds the appwrite.js to work with appwrite database management website.
in dbOperations.js we import what was configured in appwrite.js to enable CRUD functionality.
users.js, services.js, and bookings.js import from dbOperations.js and use HTTP to communicate between the database management website(appwrite) and the project files.
tests were also added

VejayPersaud committed:
Testing Fixes

VejayPersaud committed:
Database Querying and Jest Testing

Made some files to do some sample database querying and C.R.U.D operations. Additionally started some initial testing workflows.
Added a module-alias.js file to the root directory, this helps simplify problems I was having with import paths.

VejayPersaud committed:
Fixed Testing and reorganized

Babel conflicts in the root were problematic Moved the pages directory to the root

VejayPersaud committed:
Testing and Browser are running

Changed configuration files and now npm test and npm run dev are functional with no errors. next step is to properly connect the database.

buppgard committed:
Added new front end page, updated readme

VejayPersaud committed:
Merge pull request #1 from buppgard/front-end

Added new front end page, updated readme

VejayPersaud committed:
Merge branch 'OzPol:main' into dev-vpersaud1-patch-2

VejayPersaud committed:
Merge branch 'main' into dev-vpersaud1-patch-2

VejayPersaud committed:
Merge pull request #2 from VejayPersaud/dev-vpersaud1-patch-2

Dev vpersaud1 patch 2

VejayPersaud committed:
Added the React header

index.tsx cleanup

VejayPersaud committed:
Added appwrite.test.js

testing the connection to the project

OzPol reviewed (changes requested):
Can you delete babel.config.json and try again?

VejayPersaud committed:
Updated README and appwrite.test.js

tests the appwrite connection

VejayPersaud committed:
Update README.md

VejayPersaud committed:
Fixed index.tsx

fixes

VejayPersaud committed:
Merge branch 'dev-vpersaud1-patch-2' of https://github.com/VejayPersaud/ProBooker into dev-vpersaud1-patch-2

VejayPersaud committed:
Merge pull request #3 from VejayPersaud/dev-vpersaud1-patch-2

Adds the appwrite.test.js

VejayPersaud committed:
Babel Fix

installed "npm install @swc/core @swc/jest --save-dev
"
Removes babel.config.json
Adjusts jest.config.js

VejayPersaud committed:
Layout.tsx and tsconfig.json fix

Adjusts both of these so the npm run build. npm test, and npm run dev commands all work successfully.

VejayPersaud committed:
Merge pull request #4 from VejayPersaud/dev-vpersaud1-patch-babelfix

BabelFix
